Well, it turns out it wasn't that simple, of course.

gcc 16.0.1 (20260322 experimental [trunk r16-8246-g569ace1fa50]) complained with the following:

```
libgearman-server/gearmand_con.cc: In function 'gearmand_error_t gearmand_con_create(gearmand_st*, int&, const char*, const char*, gearmand_port_st*)':
libgearman-server/gearmand_con.cc:638:14: error: variable 'free_dcon_count' set but not used [-Werror=unused-but-set-variable=]
  638 |     uint32_t free_dcon_count;
      |              ^~~~~~~~~~~~~~~
  CXX      libgearman-server/libgearman_server_la-io.lo
cc1plus: all warnings being treated as errors
```

This is sort of a false positive, but not really. The `GEARMAND_LIST__DEL` macro modifies the `free_dcon_count` variable, but nothing actually references it. But I couldn't actually delete the variable entirely (I tried), not without changing that macro definition which isn't feasible since it's used all over the code. I settled on adding some `#pragma` lines to suppress the warning/error. If anyone has a better idea, let me know.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -48,6 +48,7 @@ jobs:
       GHA_CONTAINER: ${{ matrix.config.container }}
       GHA_CONFIG_NAME: ${{ matrix.config.name }}
       ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
+      DEBIAN_FRONTEND: noninteractive
 
     steps:
     - name: if running in a container, update and install sudo, git, and other basics
@@ -55,7 +56,7 @@ jobs:
       run: |
            case "${GHA_CONTAINER}" in
               ubuntu*)
-                 apt-get -o Acquire::Retries=3 update && DEBIAN_FRONTEND=noninteractive apt-get -o Acquire::Retries=3 -y install tzdata sudo apt-transport-https make apt-file software-properties-common libssl-dev build-essential autotools-dev autoconf automake pkgconf iproute2
+                 apt-get -o Acquire::Retries=3 update && apt-get -o Acquire::Retries=3 -y install tzdata sudo apt-transport-https make apt-file software-properties-common libssl-dev build-essential autotools-dev autoconf automake pkgconf iproute2
                  sudo apt-add-repository ppa:git-core/ppa
                  sudo apt-get -o Acquire::Retries=3 update && apt-get -o Acquire::Retries=3 -y install git
                  ;;
@@ -155,7 +156,7 @@ jobs:
            ./bootstrap.sh -a
     - name: configure
       run: |
-           if [[ "${CC}" == gcc* ]] && [ "${CCVERSION}" -ge 9 ]; then
+           if [[ "${CC}" == gcc* ]] && [[ "${CCVERSION}" -ge 9 ]]; then
               CXXFLAGS="-Wp,-D_GLIBCXX_ASSERTIONS"
               export CXXFLAGS
               echo "CXXFLAGS: $CXXFLAGS"
@@ -164,7 +165,7 @@ jobs:
       shell: bash
     - name: make
       run: |
-           if [[ "${CC}" == gcc* ]] && [ "${CCVERSION}" -ge 9 ]; then
+           if [[ "${CC}" == gcc* ]] && [[ "${CCVERSION}" -ge 9 ]]; then
               CXXFLAGS="-Wp,-D_GLIBCXX_ASSERTIONS"
               export CXXFLAGS
               echo "CXXFLAGS: $CXXFLAGS"

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -34,6 +34,7 @@ jobs:
           - {name: 'ubuntu-24.04 gcc-14', os: ubuntu-24.04, cc: 'gcc-14', cxx: 'g++-14', tag: '14'}
           - {name: 'ubuntu-25.04 gcc-15', os: ubuntu-latest, container: 'ubuntu:25.04', cc: 'gcc-15', cxx: 'g++-15', tag: '15', toolchain: 'ppa:ubuntu-toolchain-r/test'}
           - {name: 'ubuntu-25.10 gcc-15', os: ubuntu-latest, container: 'ubuntu:25.10', cc: 'gcc-15', cxx: 'g++-15', tag: '15'}
+          - {name: 'ubuntu-26.04 gcc-16', os: ubuntu-latest, container: 'ubuntu:26.04', cc: 'gcc-16', cxx: 'g++-16', tag: '16'}
           - {name: 'alpine-3.18 gcc-12', os: ubuntu-latest, container: 'alpine:3.18', cc: 'gcc', cxx: 'g++', tag: '12'}
           - {name: 'alpine-3.19 gcc-13', os: ubuntu-latest, container: 'alpine:3.19', cc: 'gcc', cxx: 'g++', tag: '13'}
           - {name: 'alpine-3.20 gcc-13', os: ubuntu-latest, container: 'alpine:3.20', cc: 'gcc', cxx: 'g++', tag: '13'}

--- a/libgearman-server/gearmand_con.cc
+++ b/libgearman-server/gearmand_con.cc
@@ -635,7 +635,6 @@ gearmand_error_t gearmand_con_create(gearmand_st *gearmand, int& fd,
   }
   else
   {
-    uint32_t free_dcon_count;
     gearmand_con_st *free_dcon_list= NULL;
 
     int pthread_error;
@@ -645,7 +644,6 @@ gearmand_error_t gearmand_con_create(gearmand_st *gearmand, int& fd,
 
       /* Take the free connection structures back to reuse. */
       free_dcon_list= dcon->thread->free_dcon_list;
-      free_dcon_count= dcon->thread->free_dcon_count;
       dcon->thread->free_dcon_list= NULL;
       dcon->thread->free_dcon_count= 0;
 
@@ -672,7 +670,7 @@ gearmand_error_t gearmand_con_create(gearmand_st *gearmand, int& fd,
     while (free_dcon_list != NULL)
     {
       dcon= free_dcon_list;
-      GEARMAND_LIST__DEL(free_dcon, dcon);
+      free_dcon_list= dcon->next;
       GEARMAND_LIST__ADD(gearmand->free_dcon, dcon);
     }
   }

